app/services/incorrect_profile_remover.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+# A class to remove profiles on policies that do not have a matching ref_id
+class IncorrectProfileRemover
+  def self.run!
+    counts = Policy.find_each.flat_map do |policy|
+      Profile.where(policy_object: policy)
+             .where.not(ref_id: policy.initial_profile.ref_id)
+             .destroy_all.count
+    end
+
+    Rails.logger.info "#{counts.sum} profiles removed from policies with "\
```

Comment:
For some reason this did not logged anything on `db:migrate`. Mind to change it to return the sum and then use logging in the migration (using `say` or `say_with_time`)?

Reply:
This logs to `log/development.log`, but I can make it log to STDOUT. I'm not sure what the behavior would be in production, since all logs are set to STDOUT there.

Reply:
Sorry that I'm nitpicking, why not log outside of the service or force the logging to stdout only in wrapped block:
```
# in migration
do_log_to_stdout_wrapper_if_it_exists_in_rails do
   IncorrectProfileRemover.run!
end
```
This way the logging inside of the service is preserved as configured if used outside of migration.

Reply:
If we want to change logging for all our database migrations, we can apparently set `ActiveRecord::Base.logger = Logger.new(STDOUT)`, but again, I believe migrations (and everything else) log to stdout in production already. These data migration service classes should always log to STDOUT - there's no reason why we'd want it to log anywhere else. IMO we really don't need some overly-architected logging solution here; this will only run once, after all.